### PR TITLE
Integrate Github Actions in SODA dock project 

### PR DESCRIPTION
**Pull request by ryo-osive**
**What type of PR is this?**
Integrate github action

**What this PR does / why we need it**:
Integrate github actions to SODA projects 

**Which issue(s) this PR fixes**:
#79 

**Test Report Added?**:
> /kind TESTED

**Test Report**:
image

**Special notes for your reviewer**:
Create **codecov** token enviroment variable to integrate codecov features

### CODE REVIEW

**Review by asifdxtreme — APPROVED** [review]
LGTM

**Comment by asifdxtreme** [comment]
@ryo-osive @kumarashit please add a follow up task to remove the trigger for Travis-CI and it's yml

**Review by kumarashit — APPROVED** [review]
LGTM

File: .github/workflows/build.yml [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+jobs:
+  build:
+    env:
+      GO111MODULE: on
+      TARGET: amd64
+        
+    strategy:
+#       max-parallel: 2
+      matrix:
+        go-version: [1.12.x, 1.13.x]
```
[the remark this answers is not in the document]
Comment (asifdxtreme) [reply .github/workflows/build.yml:18]: @ryo-osive yes we can remove these two..

File: .github/workflows/build.yml [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+      with:
+        go-version: ${{ matrix.go-version }}
+    
+    - name: Checkout project  
+      uses: actions/checkout@v2
+      
+    - name: Clone API project
+      run: go get github.com/sodafoundation/api
+      
+    - name: Prerequites packages on ubuntu-16.04
```
Comment (anvithks) [inline .github/workflows/build.yml:34]: Are these pre requisites installed only for Ubuntu 16.04? There are no conditional checks, hence the question.

File: .github/workflows/build.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
...
+      uses: actions/checkout@v2
+      
+    - name: Clone API project
+      run: go get github.com/sodafoundation/api
+      
+    - name: Prerequites packages on ubuntu-16.04 
+      run: sudo apt-get update && sudo apt-get install -y git make curl wget libltdl7 libseccomp2 libffi-dev gawk build-essential gcc librados-dev librbd-dev
+      
+    - name: Build
+      # Build OpenSDS Controller source code
```
Comment (anvithks) [inline .github/workflows/build.yml:38]: OpenSDS - > SODA

**Comment by AmitRoushan** [comment]
@ryo-osive 
IMO, The PR has missed job for latest golang. Is there any specific reason behind it? 

image

File: .github/workflows/build.yml (hunk 1 above)
[the remark this answers is not in the document]
Comment (ryo-osive) [reply .github/workflows/build.yml:18]: Sure, I will go the changes

**Comment by ryo-osive** [comment]
@kumarashit @anvithks 
I have updated the required changes

File: .github/workflows/build.yml [hunk 4]
```diff
@@ -0,0 +1,45 @@
...
+      with:
+        go-version: ${{ matrix.go-version }}
+    
+    - name: Checkout project  
+      uses: actions/checkout@v2
+      
+    - name: Clone API project
+      run: go get github.com/sodafoundation/api
+      
+    - name: Prerequites packages
```
Comment (anvithks) [inline .github/workflows/build.yml:34]: nit: Prerequites packages-> Install Pre-requisites
please check the typo.

File: .github/workflows/build.yml [hunk 5]
```diff
@@ -0,0 +1,45 @@
...
+      run: sudo apt-get update && sudo apt-get install -y git make curl wget libltdl7 libseccomp2 libffi-dev gawk build-essential gcc librados-dev librbd-dev
+      
+    - name: Build
+      # Build SODA Controller source code
```
Comment (anvithks) [inline .github/workflows/build.yml:38]: Is this building the SODA Controller or Dock?

File: .github/workflows/build.yml [hunk 6]
```diff
@@ -0,0 +1,45 @@
+name: Build
```
Comment (anvithks) [inline .github/workflows/build.yml:1]: Could be named as CI or something that signifies it is a CI Build.

**Review by anvithks — CHANGES_REQUESTED** [review]

**Review by kumarashit — APPROVED** [review]
LGTM

**Review by anvithks — APPROVED** [review]
Remove the WIP before merging.

**Review by asifdxtreme — APPROVED** [review]

File: .github/workflows/build.yml [hunk 7]
```diff
@@ -0,0 +1,45 @@
+name: CI Build
```
Comment (NajmudheenCT) [inline .github/workflows/build.yml:1]: should we not remove .travis.yaml when we migrate from travis to githubactions?

Comment (ryo-osive) [reply .github/workflows/build.yml:1]: That is not required to be done for getting started with github actions, yes but we should keep it in action item.

**Review by skdwriting — APPROVED** [review]
LGTM

**Pull request merged**